### ISSUE TITLE
Adding focus traversal to GUI

### ISSUE THREAD

**Issue by eschan145** [pythonarcade/arcade#1270]
In a standard GUI application, a user can switch the focus of widgets by pressing <kbd>Tab</kbd> and going back with <kbd>Shift-Tab</kbd>. It would be nice if that could be implemented in the current arcade GUI. You would need a `focus` property for each widget to see if it had focus and do something (eg. use an OpenGL filter to make it glow) and a focus property for the `UIManager`. The `focus` property in the `UIManager` could be used to find out which widget had the current focus.

Check out this [example](https://github.com/pyglet/pyglet/blob/master/examples/text/text_input.py) and try pressing <kbd>Tab</kbd> several times. Note that the focus switches between the entries. `UILabel`s should not have focus traversal. Once a widget has focus, the user could call it with a keyboard shortcut (like <kbd>Space</kbd> to dispatch the `on_press` event for buttons and <kbd>Up</kbd> and <kbd>Down</kbd> to scroll up and down in `TextArea`s.

**Comment by eruvanos** [comment]
Good call, as soon as we have a focus state in the GUI this should come next!

**Comment by eruvanos** [comment]
I am working on this now.

Focus traversal will be combined with controller support.

WIP: https://discord.com/channels/458662222697070613/716020595439173663/1336476528153727067
